src/nvim/CMakeLists.txt

```diff
@@ -189,8 +192,14 @@ list(APPEND NVIM_LINK_LIBRARIES
     ${LIBUV_LIBRARIES}
     ${MSGPACK_LIBRARIES}
     ${LUAJIT_LIBRARIES}
+    "-Wl,--whole-archive"
```

Comment:
Honestly I don't understand why this flag is necessary for libtickit & dependencies, I've used it before but only for building shared libraries that include other static libraries. Do you know a another way of avoiding missing reference errors during link stage?